### DISCUSSION TITLE
What libsum (and libmd) used for ?

### DISCUSSION THREAD

**Discussion by phidebian** [ksh93/ksh#668] [General]
Hi, 
Just curious, does someone knows what libsum is used for? with a rapid cscope I don't see any ref to it. Libsum seems to be a replacement for libmd (if libmd not present on build env) yet even no ref on libmd.

On linux (any arch) when I remove -lmd from the build ksh93 still build and pass the QA, so the question what that things used for?

Thanx in advance

**Comment by phidebian** [comment]
I figured that libsum is actually used by sum.c, cksum.c and this are part of .../lib/libcmd that raise then another question, why are we building those things, they seems not used by ksh93. They look like builtins, they defines b_sum(), b_cksum(), yet no .so are produced so it sounds like we build them but we can't load them.

Is that some archeological things?

Cheers

**Comment by tvalenta** [comment]
This came up with #596 on macOS. While it doesn't get built on macOS or FreeBSD, I presume there are still other platforms that do provide the library.

**Comment by phidebian** [comment]
Ha thanx for the pointer.

**Comment by JohnoKing** [comment]
The libsum library is used for libcmd builtins such as `cksum`, and can be enabled by setting `SHOPT ALL_LIBCMD` to 1 in `src/cmd/ksh93/SHOPT.sh` or by manually adding the builtins to the list in `src/cmd/ksh93/data/builtins.c`. Use of these is not common, though these builtins are in use on illumos (cf. https://github.com/ksh93/ksh/pull/391).
